test/frontend/mixins.spec.js

```diff
@@ -599,6 +599,71 @@ describe('Template Mixins', () => {
       });
     });
 
+    describe('maxlengthAttribute - input-text', () => {
+      it('sets maxlengthAttribute to true when explicitly set to true in field config', () => {
+        res.locals.options.fields = {
+          'field-name': {
+            maxlengthAttribute: true
+          }
+        };
+        middleware(req, res, next);
+        res.locals['input-text']().call(res.locals, 'field-name');
+        render.should.have.been.calledWith(
+          sinon.match({
+            maxlengthAttribute: true
+          })
+        );
+      });
+
+      it('sets maxlengthAttribute to false when explicitly set to false in field config', () => {
+        res.locals.options.fields = {
+          'field-name': {
+            maxlengthAttribute: false
+          }
+        };
+        middleware(req, res, next);
+        res.locals['input-text']().call(res.locals, 'field-name');
+        render.should.have.been.calledWith(
+          sinon.match({
+            maxlengthAttribute: false
+          })
+        );
+      });
+
+      it('defaults maxlengthAttribute to false when not defined in field config', () => {
+        res.locals.options.fields = {
+          'field-name': {}
+        };
+        middleware(req, res, next);
+        res.locals['input-text']().call(res.locals, 'field-name');
+        render.should.have.been.calledWith(
+          sinon.match({
+            maxlengthAttribute: false
+          })
+        );
+      });
+
+      it('sets maxlengthAttribute to false when set to a non boolean value in field config', () => {
+        const nonBooleanTypeValues = [
+          'string', null, undefined, 1234, {}, []
+        ];
+        nonBooleanTypeValues.forEach(nonBooleanValue => {
+          res.locals.options.fields = {
+            'field-name': {
+              maxlengthAttribute: nonBooleanValue
+            }
+          };
+          middleware(req, res, next);
+          res.locals['input-text']().call(res.locals, 'field-name');
+          render.should.have.been.calledWith(
+            sinon.match({
+              maxlengthAttribute: false
+            })
+          );
+        });
+      });
+    });
```

Comment:
I'm worried the forEach test for non-boolean values runs all cases in a single test. This can make it harder to pinpoint which input failed if the test breaks.

I think this makes each non-boolean case is its own test, so failures are easy to trace.
```suggestion
    // List of non-boolean values to test
      const nonBooleanTypeValues = [
        'string', null, undefined, 1234, {}, []
      ];
      nonBooleanTypeValues.forEach(nonBooleanValue => {
        it(`should set maxlengthAttribute to false when set to non-boolean value (${JSON.stringify(nonBooleanValue)}) in field config`, () => {
          res.locals.options.fields = {
            'field-name': {
              maxlengthAttribute: nonBooleanValue
            }
          };
          middleware(req, res, next);
          res.locals['input-text']().call(res.locals, 'field-name');
          render.should.have.been.calledWith(
            sinon.match({ maxlengthAttribute: false })
          );
        });
      });
    });
```